airbyte-cdk/bulk/toolkits/extract-jdbc/src/main/kotlin/io/airbyte/cdk/read/JdbcPartitionReader.kt

```diff
@@ -109,6 +123,9 @@ open class JdbcResumablePartitionReader<P : JdbcSplittablePartition<*>>(
     val runComplete = AtomicBoolean(false)
 
     override suspend fun run() {
+        // Don't start read if we've gone over max duration
+        checkMaxReadTimeElapsed()
```

Comment:
Should we also call this after our call to `coroutineContext.ensureActive()` below, in the same if block? Or is this low granularity deliberate?

Reply:
That's where I placed it initially.
The difference is that at this current location we check for elapsed duration at right _before_ read of a partition.
Moving it to after `ensureActive` means that we're checking for elapsed duration _while_ reading a partition.

If we pop out with a transient error in the middle of reading then there's no checkpoint so the work we've done for this partition before the 8 hours limit passed will be discarded.

I thought that treating the timeout not like a hard stop on the millisecond is acceptable since our partitions (chunks) should never be very long. so as long as we know to stop at the beginning of the next partition read it ok even if we go over the limit by a couple of minutes.

Reply:
OK, this is a very good reason for doing the things the way you did. You should digest this into a short comment and add it to this function call. Basically explaining that if we check during the run we're going to lose all forward progress anyway because we're not going to checkpoint.

Considering the time scales at play we shouldn't bother making this more granular. Checkpoints are always going to be every 5 or 10 minutes while the maxReadDuration is going to be an order of magnitude larger at least.